server/swap/swap_test.go

```diff
@@ -2003,18 +1994,54 @@ func TestMalformedSwap(t *testing.T) {
 	// Works with low fee, but now a confirmation.
 	tConfsSpoofer = 1
 	ensureNilErr(rig.sendSwap_maker(true))
-	rig.abcNode.invalidFeeRate = false
-	tConfsSpoofer = 0 // back to 0
+}
+
+func TestRetrySwapInit(t *testing.T) {
+	set := tPerfectLimitLimit(uint64(1e8), uint64(1e8), true)
+	matchInfo := set.matchInfos[0]
+	rig, cleanup := tNewTestRig(matchInfo)
+	defer cleanup()
+
+	rig.auth.swapReceived = make(chan struct{}, 1)
+
+	rig.swapper.Negotiate([]*order.MatchSet{set.matchSet})
+	ensureNilErr := makeEnsureNilErr(t)
+	ensureDupRequestErr := rpcErrorChecker(t, rig, msgjson.DuplicateRequestError)
+
+	ensureNilErr(rig.ackMatch_maker(true))
+	ensureNilErr(rig.ackMatch_taker(true))
+
+	ensureNilErr(rig.sendSwap_maker(true))
 
-	// Back to NewlyMatched
 	tracker := rig.getTracker()
+	// We're "rewinding time" back NewlyMatched status to be able to retry sending
+	// the same init request again.
+	tracker.mtx.Lock()
 	tracker.Status = order.NewlyMatched
-	// We're rewinding time to send the same init again. To support this hack,
-	// wait for swapSearching to clear.
-	time.Sleep(200 * time.Millisecond)
+	tracker.mtx.Unlock()
```

Comment:
Here, I still think we need to lock in order for next `processInit` to be able to read this change (it will also lock on this `tracker.mtx` for reading),

still, I couldn't reproduce this race (without locking)

Reply:
Ah well, looking at it afresh I can see why there is no race, calling `s.latencyQ.Wait` establishes "happens after" via [this chan](https://github.com/LasTshaMAN/dcrdex/blob/1201862002cab39d58b5d4366bd6e0e251367341/dex/wait/queue.go#L183-L183),

go-routine that will be working on `processInit` will have to receive on this chan first